### PR TITLE
Show unique visitors metric for all periods if available

### LINKED ISSUE #9989: Returning visits available metrics not consistent across date ranges.
When browsed for a day, this is the case:

screen shot 2016-03-31 at 16 53 13

When browsed for week, metrics are unavailable although they are still displayed:

screen shot 2016-03-31 at 16 53 33

### PR DESCRIPTION
**Pull request by sgiehl**
fixes #9989

### CODE REVIEW

File: plugins/VisitsSummary/Controller.php [hunk 1]
```diff
@@ -106,6 +107,17 @@ public function getEvolutionGraph(array $columns = array(), array $defaultColumn
...
 
+        $period = Common::getRequestVar('period', false);
+
+        if (SettingsPiwik::isUniqueVisitorsEnabled($period)) {
+            // add number of unique visitors if available
+            $selectableColumns = array_merge(
+                array($selectableColumns[0]),
+                array('nb_uniq_visitors'),
+                array_slice($selectableColumns, 1)
+            );
```
Comment (adaqus) [inline plugins/VisitsSummary/Controller.php]: If we want 'nb_uniq_visitors' to be always after 'nb_visits', maybe this is a better way (no need to modify if order of items in $selectableColumns changes):
```
$pos = array_search('nb_visits', $selectableColumns);
// array(a, b) -> array(a, x, b)
array_splice($selectableColumns, $pos+1, 0, array('nb_uniq_visitors'));
```

**Comment by adaqus** [comment]
Looks ok :+1: Take a look at the comment above.

File: plugins/VisitsSummary/Controller.php (hunk 1 above)
Comment (sgiehl) [inline plugins/VisitsSummary/Controller.php]: I see you point, but that behaviour was simply "copied" from another place.
Imho we should more likely discuss if we should introduce some kind of column ordering system that will be applied to all reports automatically. We could define that in the column classes or maybe have kind of a registry where columns can register and define there order, which would maybe make it easier for third party plugin to work with it. Maybe that could also be used to define a custom order for any user... 
@tsteur @mattab any plans to do something like that in 3.0?

Comment (tsteur) [inline plugins/VisitsSummary/Controller.php]: In Piwik 3 there is already a sparklines visualization and it is defined very differently. See eg https://github.com/piwik/piwik/blob/3.x-dev/plugins/VisitsSummary/Reports/Get.php#L164

Comment (sgiehl) [inline plugins/VisitsSummary/Controller.php]: Thanks. So, guess there is no need to solve that in 2.x aswell

**Comment by mattab** [comment]
Do we need to merge this issue?

if so, here is myreview:
*  UI tests don't show data for "Unique visitors" metric but should eg. http://builds-artifacts.piwik.org/piwik/piwik/master/19196/RowEvolution_row_evolution_other_metric.png
* XML expected artifacts should not have `unique visitors` metric, unless it is enable and/or was processed for the period.
* If we merge this we need to also fix it in Piwik 3.0 separately

**Comment by mattab** [comment]
Hi @sgiehl do you think this PR is now ready to be merged? Wondering, whether we maybe need create the PR against Piwik 3 as well?

**Comment by sgiehl** [comment]
I need to recreate the PR in order to update it. I don't think this should our could be included this way in Piwik 3, due to the changes already made there. 
We might consider merging it to 2.16.3, but I would suggest to do that after merging master to 3.x to avoid merge conflicts.
I'll recreate the PR later to check if the tests are running...

**Pull request closed without merging**